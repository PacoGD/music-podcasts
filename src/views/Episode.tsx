import { useEffect, useState } from "react";
import { useParams, Link } from "react-router-dom";
import EpisodeDetails from "../components/EpisodeDetails";
import podcastService from "../services/podcastService";

interface Episode {
  title: string;
  description: string;
  audioUrl: string;
}

function Episode() {
  const { podcastId, episodeId } = useParams<{
    podcastId: string;
    episodeId: string;
  }>();
  const [episode, setEpisode] = useState<Episode | null>(null);

  useEffect(() => {
    // FunciÃ³n para cargar los detalles del episodio desde el servicio externo.
    const loadEpisodeDetails = async () => {
      try {
        const episodeDetails: Episode | null =
          await podcastService.fetchPopularPodcasts();
        setEpisode(episodeDetails);
      } catch (error) {
        console.error("Error al cargar los detalles del episodio", error);
      }
    };

    loadEpisodeDetails();
  }, [podcastId, episodeId]);

  return (
    <div>
      <Link to={`/podcast/${podcastId}`}>Volver al podcast</Link>
      {episode ? (
        <EpisodeDetails episode={episode} />
      ) : (
        <p>Cargando detalles del episodio...</p>
      )}
    </div>
  );
}

export default Episode;
