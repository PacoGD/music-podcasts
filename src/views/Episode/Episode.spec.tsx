import { render } from "@testing-library/react";
import Episode from "./Episode";

jest.mock("../../hooks/store", () => ({
  useAppSelector: jest.fn(),
}));

describe("Episode component", () => {
  test(" snapshot", () => {
    const { container } = render(<Episode />);
    expect(container).toMatchSnapshot();
  });
});
