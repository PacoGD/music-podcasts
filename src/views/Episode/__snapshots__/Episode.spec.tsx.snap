// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode component  snapshot 1`] = `
<div>
  <div>
    <div>
      <p />
      <div />
      <audio
        controls=""
      >
        <source
          type="audio/mpeg"
        />
      </audio>
    </div>
  </div>
</div>
`;
